pkg/storagepool/v1alpha1/kubernetes_test.go

```diff
@@ -0,0 +1,420 @@
+// Copyright © 2018-2019 The OpenEBS Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package v1alpha1
+
+import (
+	"reflect"
+	"testing"
+
+	"k8s.io/apimachinery/pkg/types"
+
+	apis "github.com/openebs/maya/pkg/apis/openebs.io/v1alpha1"
+	clientset "github.com/openebs/maya/pkg/client/generated/clientset/versioned"
+	errors "github.com/openebs/maya/pkg/errors/v1alpha1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+func fakeListOk(cs *clientset.Clientset,
+	opts metav1.ListOptions) (*apis.StoragePoolList, error) {
+	return &apis.StoragePoolList{}, nil
+}
+
+func fakeListErr(cs *clientset.Clientset,
+	opts metav1.ListOptions) (*apis.StoragePoolList, error) {
+	return &apis.StoragePoolList{}, errors.New("some error")
```

Comment:
can we have `return nil, errors.New("...")`

Reply:
done